sqld/src/replication/primary/logger.rs

```diff
@@ -81,20 +73,23 @@ unsafe impl WalHook for ReplicationLoggerHook {
         };
 
         if is_commit != 0 && rc == 0 {
-            if let Some((count, checksum)) = commit_info {
-                self.commit(count, checksum);
+            if let Err(e) = self.commit() {
+                // If we reach this point, it means that we have commited a transaction to sqlite wal,
+                // but failed to commit it to the shadow WAL, which leaves us in an inconsistent state.
+                tracing::error!(
+                    "fatal error: log failed to commit: inconsistent replication log: {e}"
+                );
+                std::process::abort();
             }
```

Comment:
What can we do to restore the database if this error happens? Or is it sufficiently unlikely that we don't have to deal with it?

Reply:
Ah, I was in the middle of writing a comment in the same spirit when I noticed @honzasp just wrote it for me :innocent: Let's just make sure there's a way out of this situation, and we don't end up in a restart-abort loop

Reply:
yeah this is very unlikely. My plan is to nuke the wal, and have a mechanism to recover the wal from the db file when it's restarted. It'll allow us to load from sqlite file without dumps as well

Reply:
I opened an issue so we don't forget: https://github.com/libsql/sqld/issues/426

Reply:
Do we have any plans to merge the SQLite WAL and our shadow WAL into the same file? If we are going to do this soon, then this issue will be automatically resolved :)

Reply:
yes, sorta, but this issue will still not be completely resolved

Reply:
"soon" is also very vague at the rate things stack up on my todo 😄